docs/flink-connector.md

```diff
@@ -107,7 +105,6 @@ CREATE TABLE flink_table (
 ) WITH (
     'connector'='iceberg',
     'catalog-name'='custom_prod',
-    'catalog-type'='custom',
```

Comment:
Why we need to delete this?

Reply:
because the example as it's stated here doesn't work

Reply:
You are right, user will fail for this validation. We should delete this.

https://github.com/apache/iceberg/blob/dd5b35067baac5eb004141d51d32251d2f8e808c/flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java#L85-L90

Reply:
+1 to not having this configuration combination which can’t exist.

Should we also update the sentence on line 99 above which mentions that this “maps to the iceberg table `default_database.flink_table` managed in custom catalog.”? Specifically maybe mentioning that the table is “managed in a custom catalog of type com.my.custom.CatalogImpl” instead?

Reply:
good point, updated the sentence on L99 as you suggested